### PR TITLE
Add ability to specify tsconfig for type checked rules

### LINKED ISSUE #45: typeCheck rules don't allow specifying path to tsconfig.json
The typeCheck option added for #36 hardcodes the `tsconfig.json` as being in the `process.cwd()`, which is not the case in a hierarchy like:

```
src/
  tsconfig.json
  index.ts
webpack.config.js
```

The `tsconfig.json` is loaded in the wrong directory (ENOENT).

### PR DESCRIPTION
**Pull request by blacksonic**
resolves #45

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,14 +36,15 @@
...
     "chai": "^3.5.0",
     "enhanced-resolve": "^2.3.0",
+    "es6-promisify": "^5.0.0",
     "eslint": "3.11.1",
-    "mocha": "^3.1.2",
+    "mocha": "^3.2.0",
     "np": "^2.10.1",
     "tslint": "^4.0.0",
-    "typescript": "^2.0.10",
+    "typescript": "^2.1.1",
```
Comment (bhovhannes) [inline package.json]: Typescript 2.1.1 is not released yet, why increase the version from 2.0.10?

Comment (blacksonic) [inline package.json]: npm outdated showed it as latest version, will revert it there

**Comment by bhovhannes** [comment]
@blacksonic, will you release a new version?

**Comment by blacksonic** [comment]
Released

**Pull request merged**